docs/plots/backends.rst

```diff
@@ -0,0 +1,389 @@
+.. todo::
+   Add details
+
+Each of the plotting functions above accepts
+the following arguments: title, xlabel, ylabel, xlog, ylog, overplot, clearwindow
+
+Multi-panels plot can be set with ``set_subplot`` and ``set_jointplot``.
+See `sherpa.plot.backends.BaseBackend` for a listing of the calling signature.
+
+
+Interaction with interactive plots in the UI
+--------------------------------------------
+Each backend also acts as a context manager: All plotting commands
+in the UI are wrapped in a ``with`` statement like this::
+
+    >>> with sherpa.backend():
+    ...     plotobj.plot()
+
+That allows the backend to execute any finishing code after the plots are done,
+e.g. to save the plot to disk or to display it in a window.
+
+Other methods
+--------------
+
+Backends need to have a few more methods:
+
+- ``as_html_XXX`` (where XXX is a plot type) that are used for interactive
+  display in the notebook with ``_repr_html_``.  These functions take a plot
+  object and return a html representation as a string.
+- ``get_XXX_plot/hist_prefs`` (where XXX is a plot type) which returns a
+  dictionary of preferences that is used for displaying this plot.
+- `~sherpa.plot.backend.BasicBackend.get_latex_for_string` to format latex in strings.
+- ``colorlist(n)`` generates a list of n distinct colors. In backends that only have a
+  limited number of colors available, the list might repeat.
+
+
+How do I implement a new backend?
+=================================
+
+Example for a modified backend
+------------------------------
+Since Sherpa backends are defined using inheritance, new backends can be
+created to change the appearance of plots in ways that are beyond the
+options offered as keyword arguments for the existing backends. As an example,
+the `~sherpa.plot.pylab_backend.PylabErrorArea` backend changes the way that
+x-errors are visualized in plots from simple error bars to a shaded area.
+
+.. literalinclude:: ../../sherpa/plot/pylab_area_backend.py
+   :language: python
+   :lines: 38-99
+
+.. todo::
+
+   Put code example in here that uses it?
+
+Testing backends and plotting code
+----------------------------------
+Currently, Sherpa does not employ pixel-level tests that compare a generated
+image pixel-by-pixel to a reference image. While testing like that guarantees
+that any and all changes are found, they are susceptible to failing for
+reasons unrelated to Sherpa, such as minor changes in the default of
+:term:`matplotlib`.
+
+Instead, the plotting tests in Sherpa fall into the following categories:
+
+  - Tests that do not depend on the output of a plotting package. Those tests
+    could check e.g. the ``prepare`` stage of a plotting object.
+    They work with any backend (since Sherpa has `sherpa.plot.backends.BasicBackend`
+    there is always a backend). For those tests, no special
+    setup is needed and they will just be run with whatever backend is loaded
+    based on the installed packages and the `.sherpa.rc` file.
```

Comment:
```suggestion
    based on the installed packages and the ``.sherpa.rc`` file.
```

With a single backtick it seemed to lose the leading "." in the HTML sphinx generated. There is a long-standing desire by me to have a "good" way to say "configuration file" - with an appropriate link - but hat's for another year.